pkg/infra/webhooks.go

```diff
@@ -0,0 +1,179 @@
+package infra
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-api-actuator-pkg/pkg/framework"
+	gcp "github.com/openshift/cluster-api-provider-gcp/pkg/apis/gcpprovider/v1beta1"
+	mapiv1 "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	vsphere "github.com/openshift/machine-api-operator/pkg/apis/vsphereprovider/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	aws "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsprovider/v1beta1"
+	azure "sigs.k8s.io/cluster-api-provider-azure/pkg/apis/azureprovider/v1beta1"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("[Feature:Machines] Webhooks", func() {
+	var client runtimeclient.Client
+	var platform configv1.PlatformType
+	var machineSetParams framework.MachineSetParams
+
+	var delObjects map[string]runtime.Object
+
+	var ctx = context.Background()
+
+	BeforeEach(func() {
+		delObjects = make(map[string]runtime.Object)
+
+		var err error
+		client, err = framework.LoadClient()
+		Expect(err).ToNot(HaveOccurred())
+
+		// Only run on platforms that have webhooks
+		clusterInfra, err := framework.GetInfrastructure(client)
+		Expect(err).NotTo(HaveOccurred())
+		platform = clusterInfra.Status.PlatformStatus.Type
+		switch platform {
+		case configv1.AWSPlatformType, configv1.AzurePlatformType, configv1.GCPPlatformType, configv1.VSpherePlatformType:
+			// Do Nothing
+		default:
+			Skip(fmt.Sprintf("Platform %s does not have webhooks, skipping.", platform))
+		}
+
+		machineSetParams = framework.BuildMachineSetParams(client, 1)
+		ps, err := createMinimalProviderSpec(platform, machineSetParams.ProviderSpec)
+		Expect(err).ToNot(HaveOccurred())
+		machineSetParams.ProviderSpec = ps
+	})
+
+	AfterEach(func() {
+		Expect(deleteObjects(client, delObjects)).To(Succeed())
+	})
+
+	It("should be able to create a machine from a minimal providerSpec", func() {
+		machine := &mapiv1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: fmt.Sprintf("%s-webhook-", machineSetParams.Name),
```

Comment:
can we prefix with `clusterInfra.Status.InfrastructureName`. This is used by installer/CI to garbage collect on destroy.

Reply:
It's using the name from the `machineSetParams` which was built by `BuildMachineSetParams`, so it should have that already, it should be `<clusterInfra.Status.InfrastructureName>-webhook-<rand>` when it's created